### ISSUE TITLE
heatmap 

### ISSUE THREAD

**Issue by SaidaSaad** [jessegmeyerlab/positional-SHAP#2]
Dear All,
Thank you so much .
I have classification problem for protein sequences .
So My input data is 1D sequence of length 340 (17 amino acids (each 21 hot vector ) ) and output three classes.
I have my model. I would like to ask on if it is possible to have heat map and shap analysis to know which part of the sequence contributes , My problem is that how  I can define the features for shap so I can have plot since I train the model with only 
My training set is like one dimension so I have only one feature.

**Comment by qdickinson** [comment]
Hello Saida,

Using one dimensional data might be a challenge since SHAP assigns it's values to the input features themselves. Ideally for PoSHAP to work, you would split your sequence into 340 input features, one for each position. Is your model able to be set up in that manner?

**Comment by SaidaSaad** [comment]
Hi Quinn

Thanks for your reply.
I have protein sequences each of length 17 amino acids. I preprocess by map to one hot encoding for each then I flatten then into 340 (17 *21 = 340 ) , I gave it to the model 340 as single input . and get one output (class) .

**Comment by qdickinson** [comment]
Thanks, that makes it more clear. I haven't run SHAP with a flattened input yet, so I'm unsure what it will return. It may give you the importances of each of the indices of your input array. If that's the case, you should be able to determine the importance of each position by reversing the flattening and encoding. For example, if you had a sequence length 2, with 3 amino acids, your encoded flattened array should look something like [0,0,1,1,0,0] for c-a with the amino acids being represented by a=[1,0,0], b=[0,1,0], and c=[0,0,1]. So if your shap array looked like [0.3, 0.5, 1, 3, -0.5, 2], you could say the contribution of it being a c at position 1 is 1.8 and a at position 2 is 4.5. But this would require some data manipulation to make it work.

PoSHAP should work as is if you do not flatten your input. You would then have 17 inputs to your model and a dictionary linking each encoding to an amino acid. It might be worth trying to see how this change effects your model's accuracy.

**Comment by SaidaSaad** [comment]
Thank you so much for your reply Quinn , it really helpful and clear .
it ijust takes me time to understand how to do that in my model. Also another challenge is that i use pytorch. But I have question you mean instead of giving 1 vector of length 340 . I have list of lists right?

**Comment by qdickinson** [comment]
I believe that's correct. It would be input something like  [[0,0,1],[1,0,0]] rather than [0,0,1,1,0,0]. You would have a dictionary then relating your encoding to each input amino acid.
